Library/Homebrew/test/spec_helper.rb

```diff
@@ -166,6 +171,23 @@
     skip "Requires network connection." unless ENV["HOMEBREW_TEST_ONLINE"]
   end
 
+  config.before do |example|
+    next if example.metadata.key?(:needs_network)
+    next if example.metadata.key?(:allow_utils_curl)
+
+    DISALLOWED_UTILS_CURL_METHODS.each do |method|
+      allow(Utils::Curl).to receive(method).and_raise(<<~ERROR)
+        Unexpected call to Utils::Curl.#{method} without setting :needs_network.
+
+        If this curl method never makes network requests, add it to the
+        ALLOWED_UTILS_CURL_METHODS array in the spec_helper.rb file.
+
+        If this curl method won't make network requests this time in particular,
+        set :allow_utils_curl on the failing spec.
```

Comment:
Least helpful comment ever: can't really explain why but this kinda feels weird to me. Would it be possible to make this more specific so e.g. it only fails if `curl` itself is actually executed?